quantize.py

```diff
@@ -212,8 +206,12 @@ def get_max_bit_shift(t, return_bit_shift=False):
                 # and is therefore always 128.
                 weights *= 2**(tc.dev.ACTIVATION_BITS-1)
 
-                if first_layer:  # Never True for QAT
-                    weights *= 2
+                if first:  # Only True for QAT
+                    weights *= 2.  # Multiply first layer by 2. FIXME: Address this in DataLoader
+
+                # Ensure it fits and is an integer
+                weights = weights.add(.5).floor().clamp(min=-(2**(clamp_bits-1)-lower_bound),
```

Comment:
1) clamp limit should be 2**(tc.dev.ACTIVATION_BITS-1) times what is written. Because we multiplied it with term at line 207. 

2) lower_bound needs to be removed if unnecessary, same applies in the weights too.

Reply:
See commit d462046